src/smt/smt_engine.cpp

```diff
@@ -1214,6 +1214,10 @@ void SmtEngine::setDefaults() {
   {
     d_logic = LogicInfo("QF_BV");
   }
+  else if (d_logic.getLogicString() == "QF_UFBV" && options::ackermann())
+  {
+    d_logic = LogicInfo("QF_BV");
+  }
```

Comment:
If I am understanding correctly, this should apply to all logics that include `UF`. So it should be something like:
```
if (d_logic.isTheoryEnabled(THEORY_UF) && options::ackermann()) {
  d_logic.disableTheory(THEORY_UF);
}
```